lib/content-api.js

```diff
@@ -60,6 +73,21 @@ function transformData(obj, dirpath) {
 }
 
 ContentApi.prototype.getRecord = function getRecord(slug) {
+  return new Promise((resolve) => {
+    this.getRecordFromApi(slug)
+      .then((record) => {
+        logger.verbose(`Loaded from content API: ${slug}`);
+        resolve(record);
+      })
+      .catch((error) => {
+        logger.error(error);
```

Comment:
Re. the promise yes it's a good idea.

I would also try a different design and see if you like it:
- have `ContentApiHandler`s being just a function taking `slug` as param
- have 2 different ContentAPIHandlers: FromFile and FromAPI
- add to the `ContentAPI` lib (the one that you have) a list of handlers
- have the `ContentApi.prototype.getRecord` method loop over the handlers and return the first resolved item

This is so that we can:
- test the handlers independently
- configure the handlers easily by just changing the list